app/code/Magento/Catalog/Controller/Adminhtml/Category/Add.php

```diff
@@ -61,9 +72,8 @@ public function execute()
             $category->addData($categoryData);
         }
 
-        $resultPageFactory = $this->_objectManager->get(\Magento\Framework\View\Result\PageFactory::class);
-        /** @var \Magento\Backend\Model\View\Result\Page $resultPage */
-        $resultPage = $resultPageFactory->create();
+        /** @var Page $resultPage */
+        $resultPage = $this->resultPageFactory->create();
```

Comment:
Hi @drpayyne,
AFAIK there should be available factory for all types of results - `$this->resultFactory`. Is so - we should use it instead of introducing new dependencies to constructor.
Could you check that?

Reply:
https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Controller/ResultFactory.php

Reply:
Fixed the code according to your requested changes. Thanks!